src/analysis/bounded_variation.lean

```diff
@@ -627,6 +628,26 @@ begin
   exact comp_le_of_monotone_on _ ψ hψ ψts,
 end
 
+lemma comp_eq_of_monotone_on_inter_Icc (f : α → E) {s : set α} {t : set β} (φ : β → α)
+  (hφ : monotone_on φ t) (φst : set.maps_to φ t s) (φsur : set.surj_on φ t s) {x y : β}
+  (hx : x ∈ t) (hy : y ∈ t) (h : x ≤ y) :
```

Comment:
Isn't it true even without `h`?

Reply:
Thanks, it is! But the proof is a bit ugly since I'm now copy-pasting a snippet amounting to `x≥y -> (Icc x y).subsingleton`.
I'd rather not have to touch a synchronized file to get this in, but could if you think it's necessary.

Reply:
Maybe make it a local lemma, as they propose here: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F

Reply:
Yes, make it a lemma in the current file with a comment like `porting note: move to file ... once the port is over`. (And check whether the file ... has indeed already be ported or not).